likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js

```diff
@@ -0,0 +1,214 @@
+
+  const { postUserLikerInfo } = useDispatch(USER_LIKER_INFO_STORE_NAME);
+
+  const [siteLikerIdEnabled, enableSiteLikerId] = useState(DBSiteLikerIdEnabled);
+  // If siteLikerId is enabled === not editable,
+  // then overwrite the user liker info with site liker info
+  const defaultLikerId = siteLikerIdEnabled
+    ? DBSiteLikerId
+    : DBUserLikerId;
+  const defaultLikerDisplayName = siteLikerIdEnabled
+    ? DBSiteLikerDisplayName
+    : DBUserLikerDisplayName;
+  const defaultLikerWalletAddress = siteLikerIdEnabled
+    ? DBSiteLikerWallet
+    : DBUserLikerWallet;
+  const defaultLikerAvatar = siteLikerIdEnabled
+    ? DBSiteLikerAvatar
+    : DBUserLikerAvatar;
+  const [likerIdValue, getLikerIdValue] = useState(defaultLikerId);
+  const [likerDisplayName, getLikerDisplayName] = useState(
+    defaultLikerDisplayName,
+  );
+  const [likerWalletAddress, getLikerWalletAddress] = useState(
+    defaultLikerWalletAddress,
+  );
+  const [likerAvatar, getLikerAvatar] = useState(defaultLikerAvatar);
+  const [savedSuccessful, setSavedSuccessful] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [, setIsDisconnect] = useState(false);
+  const [isChangingTypingLiker, setIsChangingTypingLiker] = useState(false);
+  const [hasValidLikecoinId, setHasValidLikecoinId] = useState(false);
+  const [showChangeButton, setShowChangeButton] = useState(true);
+  const [showDisconnectButton, setShowDisconnectButton] = useState(false);
+
+  // Update Data based on data returned by likecoin server.
+  const fetchLikeCoinID = useMemo(
+    () => debounce(async (likerId) => {
+      setSavedSuccessful(false);
+      if (!likerId) return;
+      try {
+        const response = await axios.get(
+          `https://api.like.co/users/id/${likerId}/min`,
+        );
+        getLikerIdValue(response.data.user);
+        getLikerDisplayName(response.data.displayName);
+        getLikerWalletAddress(response.data.cosmosWallet);
+        getLikerAvatar(response.data.avatar);
+        setIsLoading(false);
+        setHasValidLikecoinId(true);
+      } catch (error) {
+        setIsLoading(false);
+        getLikerIdValue('');
+        getLikerDisplayName('');
+        getLikerWalletAddress('');
+        getLikerAvatar('');
+        setHasValidLikecoinId(false);
+      }
+    }, 500),
+    [],
+  );
+  useEffect(() => {
+    fetchLikeCoinID(likerIdValue);
+  }, [fetchLikeCoinID, likerIdValue]);
+  useEffect(() => {
+    enableSiteLikerId(DBSiteLikerIdEnabled);
+    getLikerIdValue(defaultLikerId);
+    getLikerDisplayName(defaultLikerDisplayName);
+    getLikerWalletAddress(defaultLikerWalletAddress);
+    getLikerAvatar(defaultLikerAvatar);
+    setShowChangeButton(!!defaultLikerId);
+    setShowDisconnectButton(!!defaultLikerId);
+    setHasValidLikecoinId(!!defaultLikerId);
+  }, [
+    DBSiteLikerIdEnabled,
+    defaultLikerId,
+    defaultLikerDisplayName,
+    defaultLikerWalletAddress,
+    defaultLikerAvatar,
+  ]);
```

Comment:
`useState` already comes with a default value that you set, what are these lines for?

Reply:
To change dynamically when default values changes. But how default value changes is confusing. I will re-write that part.